lux/media/lux/lux.js

```diff
@@ -3430,58 +3447,427 @@ angular.module('lux.form.process', ['ngFileUpload'])
+                                    }
+                                });
+                            }
+                        }
+                    });
+
+                    defer.resolve(data);
+
+                }, function(data) {
+                    options[0] = '(error loading options)';
+                    defer.reject();
+                });
+                return defer.promise;
+            },
+
+            /**
+             * Parses single option to the correct format.
+             *
+             * @param option {object} - id and repr attributes
+             * @param attrs {object}
+             * @param config {object} - parameters passed to query string in request
+             * @returns {object}
+             */
+            parseOption: function(option, attrs, config) {
+                var parsedOption = {
+                    id: option[config.id],
+                    repr: option[config.nameOpts.source]
+                };
+
+                if (config.nameFromFormat) {
+                    parsedOption.repr = formatString(config.nameOpts.source, option);
+                }
+
+                if (attrs.multiple) {
+                    // For multiple field always get id of the object
+                    parsedOption.id = option.id;
+                }
+
+                return parsedOption;
+            },
+
+            /**
+             * Returns all parsed options.
+             *
+             * @param raw_options {object} - options fetched from the API
+             * @param options {object} - current `options` from select
+             * @param attrs {object}
+             * @param config {object} - parameters passed to query string in request
+             * @param extendCurrentOptions {boolean} - indicates whether to extend current `options` using `raw_options` from the API
+             */
+            getOptions: function(raw_options, options, attrs, config, extendCurrentOptions) {
+                angular.forEach(raw_options, function (option) {
+                    var parsedOption = remoteService.parseOption(option, attrs, config);
+
+                    if (extendCurrentOptions) {
+                        if (remoteService.excludeValue !== parsedOption.id) {
+                            options.push(parsedOption);
+                        }
+                    } else {
+                        options.push(parsedOption);
+                    }
+                });
+            }
+        };
+
+        return remoteService;
+    }])
+    /**
+     * Extension of ui-select to support infinite list of items.
+     */
+    .directive('selectInfinity', ['$parse', '$timeout', function($parse, $timeout) {
+        /**
+         * Returns the height of the given element
+         *
+         * @param elem
+         * @returns {integer} - height of the element
+         */
+        function height(elem) {
+            elem = elem[0] || elem;
+            if (isNaN(elem.offsetHeight)) {
```

Comment:
What is the isNaN check here for? isNaN has odd results because it coerces the argument to a number first. 
When is `elem.document` defined?